test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -273,14 +295,26 @@ older_release_branches=(mongodb-4.2 mongodb-4.0 mongodb-3.6)
 
 # This array is used to generate compatible configuration files between releases, because
 # upgrade/downgrade test runs each build's format test program on the second build's
-# configuration file. 
+# configuration file.
 compatible_upgrade_downgrade_release_branches=(mongodb-4.4 mongodb-4.2)
 
 declare -A scopes
 scopes[newer]="newer stable release branches"
 scopes[older]="older stable release branches"
 scopes[wt_standalone]="WiredTiger standalone releases"
 
+# The following associative array maps the 'official' build system to use for each branch.
+# CMake build support is reliably mature in newer release branches, whilst earlier revisions
+# primarily use Autoconf (note: some earlier branches may have CMake support, but these aren't
+# considered 'mature' versions.)
+declare -A build_sys
```

Comment:
I've chosen the cut-off for CMake testing to be `develop` and any future release branches we add to this script, i.e. CMake can be considered the official build system in any releases from _this_ point in time.  
Even though `mongodb-5.0` and `mongodb-4.4` technically have a CMake build, I feel strongly its better to test these branches with `autoconf` since:
1.  The main build system used at the time of testing these release was `autoconf`(being the 'official' build system for those release).
2. CMake was not completely 'mature' in these brancheas. Several improvements & refactors have since been made (particularly in the lead up to this ticket) that aren't present in these releases.

Reply:
Sounds like a well-thought approach to me. Once this work is done, let's think about whether we'd like to have older branches adopt CMake-only builds. And if yes, how to handle backports.